dracut/repartition/endless-repartition.sh

```diff
@@ -39,14 +36,28 @@ if [ -z ${root_part} ]; then
   exit 0
 fi
 
+get_last_char() {
+	ret=$1
+	while [ "${#ret}" != "1" ]; do
+		ret=${ret#?}
+	done
+	echo $ret
+}
+
+# We have to work with a variety of setups here:
+# - Either SCSI or MMC block devices with different naming conventions
+# - A different partition numbering scheme as not all configurations have
+#   a ESP and a BIOS boot partition.
+partno=$(get_last_char ${root_part})
```

Comment:
If this is bash, you can just use the offset expansion with a negative offset of 1 - `${root_part: -1}`. If this isn't bash, the above function is fine.